### PR TITLE
 Add HTTP to HTTPS redirect

### PR DESCRIPTION
**Pull request by gene1wood**
* Create a new CloudFront distribution so that the site will listen for http port 80 requests to redirect them to https (this isn't possible with API Gateway)
* Change the case of the http header names from all lower case to the first letter being uppercase as this changes when going from API Gateway -> Lambda to CloudFront -> API Gateway -> Lambda
* Move deployment from us-west-2 to us-east-1 to accomodate CloudFront
* Add a check in deploy.sh to prevent deploying in any region other than us-east-1
* Change ACM cert for prod to new us-east-1 based cert
* Create new ACM cert for dev (the ACM cer ARN in there before was for an unrelated cert)
* Remove the custom domain name logic from API Gateway as the domain name now will belong to CloudFront and API Gateway will have no custom domain name
* Set http headers to prevent CloudFront from caching

In summary to add http to https redirect to an API Gateway fronted Lambda function you must
* Move the deployment and ACM certs to us-east-1
* Add a CloudFront distribution to the stack
* Modify the Lambda logic to deal with http header names with the first character being uppercase
* Enumerate the HTTP headers you need to have passed through

# Other changes
*  Remove remaining S3 settings and permissions
  * These were leftover after changing from an S3 data store to user cookies in 148a24f
*  Show CloudFormation stack events after a failure during deployment
*  Set loglevel for utils

### CODE REVIEW

File: federated-aws-rp.yaml [hunk 1]
```diff
@@ -149,23 +117,14 @@ Resources:
...
   AwsFederatedRpRoute53RecordSet:
     Type: AWS::Route53::RecordSet
     Condition: UseCustomDomainName
     Properties:
       AliasTarget:
-        DNSName: !GetAtt AwsFederatedRpDomainName.RegionalDomainName
-        HostedZoneId: !GetAtt AwsFederatedRpDomainName.RegionalHostedZoneId
-      Comment: Bind the custom domain name to the AwsFederatedRp API Gateway
+        DNSName: !GetAtt AwsFederatedRpCloudFrontDistribution.DomainName
+        HostedZoneId: Z2FDTNDATAQYW2  # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-route53-aliastarget-1.html
```
Comment (andrewkrug) [inline federated-aws-rp.yaml:126]: Maybe break this out to a default param for reusability.

Comment (gene1wood) [reply federated-aws-rp.yaml:126]: @andrewkrug This is actually a global hard coded value (see that link in the comment) and should always be the same for all AWS customers

**Review by andrewkrug — APPROVED** [review]
